conans/client/cmd/new.py

```diff
@@ -281,14 +311,29 @@ def cmd_new(ref, header=False, pure_c=False, test=False, exports_sources=False,
         files = {"conanfile.py": conanfile_bare.format(name=name, version=version,
                                                        package_name=package_name)}
     elif template:
-        if not os.path.isabs(template):
-            template = os.path.join(cache.cache_folder, "templates", template)
-        if not os.path.isfile(template):
-            raise ConanException("Template doesn't exist: %s" % template)
-        conanfile_template = load(template)
-        t = Template(conanfile_template)
-        replaced = t.render(name=name, version=version, package_name=package_name)
-        files = {"conanfile.py": replaced}
+        is_file_template = os.path.basename(template).endswith('.py')
+        if is_file_template:
+            if not os.path.isabs(template):
+                old_path = os.path.join(cache.cache_folder, "templates", template)
+                new_path = os.path.join(cache.cache_folder, "templates", "cmd_new", template)
+                template = new_path if os.path.isfile(new_path) else old_path
+            if not os.path.isfile(template):
+                raise ConanException("Template doesn't exist: %s" % template)
+            replaced = _render_template(load(template), 
+                                        name=name, 
+                                        version=version, 
+                                        package_name=package_name)
+            files = {"conanfile.py": replaced}
+        else:
```

Comment:
If it is a folder, I think it should be incompatible with other arguments, at least with `test` and `pure-c` too. Other arguments might be deprecated in the new CLI implementation.

Reply:
> If it is a folder, I think it should be incompatible with other arguments, at least with test and pure-c too

`pure-c` should be incompatible with `--template=FILE` as well (judging by the code). Will be amended.

Not sure about the `test` though. I mean a template might not provide a `test_package` and only provide recipe and project structure. Maybe just add a check that there is no `test_package` files already (and error out if that's not the case)?

Reply:
Same for `.gitignore`: template *might* provide this file, but it might not as well.

Reply:
Ok, leave it as it is. The new CLI will need to decide about all these arguments. There are too many flags here.

Reply:
I've added a check for `pure-c` though =) (since it's ignored with `template` anyways)